v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+    {
+      id:
+        message.properties.product_id ||
+        message.properties.id ||
+        message.properties.sku ||
+        "",
+      quantity: message.properties.quantity,
+      item_price: message.properties.price
+    }
+  ];
+  return {
+    content_ids: contentIds,
+    content_type: contentType,
+    content_name: contentName,
+    content_category: contentCategory,
+    currency,
+    value,
+    contents
+  };
+}
+
+function checkPiiProperties(
+  message,
+  custom_data,
+  blacklistPiiProperties,
+  whitelistPiiProperties,
+  isStandard,
+  eventCustomProperties
+) {
+  const defaultPiiProperties = [
+    "email",
+    "firstName",
+    "lastName",
+    "firstname",
+    "lastname",
+    "first_name",
+    "last_name",
+    "gender",
+    "city",
+    "country",
+    "phone",
+    "state",
+    "zip",
+    "birthday"
+  ];
+  blacklistPiiProperties = blacklistPiiProperties || [];
+  whitelistPiiProperties = whitelistPiiProperties || [];
+  const customPiiProperties = {};
+  for (let i = 0; i < blacklistPiiProperties.length; i += 1) {
+    const configuration = blacklistPiiProperties[i];
+    customPiiProperties[configuration.blacklistPiiProperties] =
+      configuration.blacklistPiiHash;
+  }
+  Object.keys(custom_data).forEach(function(property) {
+    const isPropertyPii = defaultPiiProperties.indexOf(property) >= 0;
+    let isProperyWhiteListed = false;
+    for (let i = 0; i < whitelistPiiProperties.length; i += 1) {
+      const configuration = whitelistPiiProperties[i];
+      const properties = configuration.whitelistPiiProperties;
+      if (properties === property) {
+        isProperyWhiteListed = true;
+      }
+    }
+    if (isPropertyPii) {
+      if (!isProperyWhiteListed) {
+        delete custom_data[property];
+      }
+    }
+
+    if (Object.prototype.hasOwnProperty.call(customPiiProperties, property)) {
+      if (customPiiProperties[property]) {
+        custom_data[property] = sha256(String(message.properties[property]));
+      } else {
+        delete custom_data[property];
+      }
+    }
+    let isCustomProperty = false;
+    for (let i = 0; i < eventCustomProperties.length; i += 1) {
+      const configuration = eventCustomProperties[i];
+      const properties = configuration.eventCustomProperties;
+      if (properties === property) {
+        isCustomProperty = true;
+      }
+    }
+    if (isStandard && !isCustomProperty && !isPropertyPii) {
+      delete custom_data[property];
+    }
+  });
+
+  return custom_data;
+}
+
+function responseBuilderSimple(message, category, destination) {
+  const { Config } = destination;
+  const { pixelId, accessToken } = Config;
+  const {
+    blacklistPiiProperties,
+    categoryToContent,
+    eventCustomProperties,
+    valueFieldIdentifier,
+    whitelistPiiProperties,
+    limitedDataUSage
+  } = Config;
+
+  const endpoint = `https://graph.facebook.com/v9.0/${pixelId}/events?access_token=${accessToken}`;
+
+  const user_data = constructPayload(
```

Comment:
Any specific reason for not using camelCase?

Reply:
yeah they accept as user_data key